lib/util.js

```diff
@@ -672,22 +671,33 @@ function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
   let visibleLength = 0;
   let index = 0;
-  while (index < value.length && visibleLength < ctx.maxArrayLength) {
-    let emptyItems = 0;
-    while (index < value.length && !hasOwnProperty(value, String(index))) {
-      emptyItems++;
-      index++;
-    }
-    if (emptyItems > 0) {
+  for (const elem in value) {
+    if (visibleLength === ctx.maxArrayLength)
+      break;
+    const i = +elem;
+    if (index !== i) {
+      if (i > 0 === false)
```

Comment:
Can you add a comment on why this is necessary, and different from `i <= 0`? (I understand it because I saw an earlier version of the PR, but comments are more for future than now anyway.)